File: client/samples/Python/sequoiadb/connect.py
/*******************************************************************************

   Copyright (C) 2011-Present SequoiaDB Ltd.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.

   Source File Name = connect.py

   Descriptive Name = N/A

   Dependencies: N/A

   Restrictions: N/A

   Change Activity:
   defect Date        Who         Description
   ====== =========== =========== =============================================
          23/07/2025  fangjiabin  Initial Draft

   Last Changed =

*******************************************************************************/
#! /usr/bin/python

import pysequoiadb
from pysequoiadb import client
from pysequoiadb.error import (SDBTypeError,
                               SDBBaseError,
                               SDBEndOfCursor)

if "__main__" == __name__:

   # connect to local db, using default args value.
   # host= 'localhost', port= 11810, user= '', password= ''
   try:
      db_default = client()
      print( db_default )
      del db_default
   except (SDBTypeError, SDBBaseError) as e:
      print(e)

   # connect to db, using default args value.
   # host= '192.168.20.111', port= 11810, user= '', password= ''
   # 192.168.20.111 is not a valid 
   try:
      db_to_1 = client('192.168.20.48', 11810, '', '')
      del db_to_1
   except (SDBTypeError, SDBBaseError) as e:
      print(e)

   # connect to db, using default args value.
   # host= 'localhost', port= 11810, user= '', password= ''
   try:
      db = client("ubuntu-dev9", 11810)

      host = "192.168.20.48"
      service = '11860'

      # try to connect another db server by service
      db.connect(host, service)
      db.disconnect()
      # try to connect other db server
      hosts = [ {'host':'192.168.20.48', 'service':11810},
                {'host':'192.168.20.111', 'service':50000},
                {'host':'localhost', 'service':11810} ]
      db.connect_to_hosts(hosts, user="", password="")

      # close connection to db server
      db.disconnect()

      # release clinet
      del db

   except (SDBTypeError, SDBBaseError) as e:
      print(e)

File: client/samples/CPP/connect.cpp
/*******************************************************************************

   Copyright (C) 2011-Present SequoiaDB Ltd.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.

   
*******************************************************************************/
#include <iostream>
#include "common.hpp"

using namespace std ;
using namespace sdbclient ;
using namespace sample ;

// Display Syntax Error
void displaySyntax ( CHAR *pCommand ) ;

INT32 main ( INT32 argc, CHAR **argv )
{
  // verify syntax
  if ( 5 != argc )
  {
     displaySyntax ( (CHAR*)argv[0] ) ;
     exit ( 0 ) ;
  }
  // read argument
  CHAR *pHostName    = (CHAR*)argv[1] ;
  CHAR *pPort        = (CHAR*)argv[2] ;
  CHAR *pUsr         = (CHAR*)argv[3] ;
  CHAR *pPasswd      = (CHAR*)argv[4] ;

  // define local variable
  sdb connection ;
  INT32 rc = SDB_OK ;

  // connect to database
  rc = connection.connect( pHostName, pPort, pUsr, pPasswd ) ;
  if( rc!=SDB_OK )
  {
     cout<<"Fail to connet to database, rc = "<<rc<<endl ;
     goto error ;
  }
  else
     cout<<"Connect success!"<<endl ;

done:
  // disconnect from database
  connection.disconnect () ;
  return 0 ;
error:
  goto done ;
}

// Display Syntax Error
void displaySyntax ( CHAR *pCommand )
{
  cout << "Syntax:" << pCommand 
       << " <hostname> <servicename> <username> <password>" << endl ;
}

